packages/client-core/src/admin/components/settings/tabs/instanceServer.tsx

```diff
@@ -53,50 +55,132 @@ const InstanceServerTab = forwardRef(({ open }: { open: boolean }, ref: React.Mu
+    return engineSettings.find((setting) => setting.key === settingName)?.value || ''
+  }
 
-  const settingsState = useHookstate(null as null | InstanceServerSettingType)
+  const domainValue = getSettingValue(EngineSettings.InstanceServer.Domain)
+  const clientHostValue = getSettingValue(EngineSettings.InstanceServer.ClientHost)
+  const rtcStartPortValue = getSettingValue(EngineSettings.InstanceServer.RtcStartPort)
+  const rtcEndPortValue = getSettingValue(EngineSettings.InstanceServer.RtcEndPort)
+  const rtcPortBlockSizeValue = getSettingValue(EngineSettings.InstanceServer.RtcPortBlockSize)
+  const identifierDigitsValue = getSettingValue(EngineSettings.InstanceServer.IdentifierDigits)
+  const releaseNameValue = getSettingValue(EngineSettings.InstanceServer.ReleaseName)
+  const portValue = getSettingValue(EngineSettings.InstanceServer.Port)
+  const modeValue = getSettingValue(EngineSettings.InstanceServer.Mode)
+  const locationNameValue = getSettingValue(EngineSettings.InstanceServer.LocationName)
+  const webRTCSettingsValue = getSettingValue(EngineSettings.InstanceServer.WebRTCSettings) || '{}'
 
   useEffect(() => {
-    if (instanceServerSettings) {
-      settingsState.set(instanceServerSettings)
+    if (engineSettings.length > 0) {
+      domainState.set(domainValue)
+      clientHostState.set(clientHostValue)
+      rtcStartPortState.set(rtcStartPortValue)
+      rtcEndPortState.set(rtcEndPortValue)
+      rtcPortBlockSizeState.set(rtcPortBlockSizeValue)
+      identifierDigitsState.set(identifierDigitsValue)
+      releaseNameState.set(releaseNameValue)
+      portState.set(portValue)
+      modeState.set(modeValue)
+      locationNameState.set(locationNameValue)
+      webRTCSettingsState.set(JSON.parse(webRTCSettingsValue))
       state.set({ loading: false, errorMessage: '' })
     }
-  }, [instanceServerSettings])
+  }, [engineSettings])
 
   const handleSubmit = (event) => {
     state.loading.set(true)
     event.preventDefault()
-    const newSettings = {
-      ...settingsState.get(NO_PROXY),
-      local: Boolean(settingsState.value?.local),
-      createdAt: undefined!,
-      updatedAt: undefined!
-    } as any as InstanceServerSettingType
-
-    API.instance
-      .service(instanceServerSettingPath)
-      .patch(id, newSettings)
+    const setting = {
+      clientHost: clientHostState.value,
+      rtcStartPort: parseInt(rtcStartPortState.value),
+      rtcEndPort: parseInt(rtcEndPortState.value),
+      rtcPortBlockSize: parseInt(rtcPortBlockSizeState.value),
+      identifierDigits: parseInt(identifierDigitsState.value),
+      domain: domainState.value,
+      releaseName: releaseNameState.value,
+      port: parseInt(portState.value),
+      mode: parseInt(modeState.value),
+      locationName: parseInt(locationNameState.value),
+      WebRTCSettings: JSON.stringify(webRTCSettingsState.value),
+      local: Boolean(getSettingValue(EngineSettings.InstanceServer.Local))
+    }
```

Comment:
Please reduce the code here and make it somewhat generic.

Reply:
done